bin/index.js

```diff
@@ -1,77 +1,43 @@
 
-packageInfo = JSON.parse(readFileSync('./package.json'));
-settings = packageInfo['airtable-schema-generator'];
-let inputFolder = '';
-let outputFolder = '';
-let AIRTABLE_BASE_ID = '';
-let DEFAULT_VIEW = 'Grid view';
-if (settings && settings.output) {
-  inputFolder = settings.input || '.';
-  outputFolder = settings.output;
-  AIRTABLE_BASE_ID = settings.baseId || '';
-  DEFAULT_VIEW = settings.defaultView || 'Grid view';
-  main();
-} else {
-  console.log("Couldn't find Input and Output Folder Path in Settings Object:");
-  console.log(settings);
-  console.log(
-    "Please add appropriate values for 'input' and 'output' to package.json under 'airtable-schema-generator' key."
-  );
-}
-
-async function main() {
-  // Use Electron to fetch schema from API Docs
-  // let schema = await fetchSchema({
-  //   email: process.env.AIRTABLE_EMAIL,
-  //   password: process.env.AIRTABLE_PASSWORD,
-  //   baseId: process.env.AIRTABLE_BASE_ID
-  // });
+function readSettings() {
+  const packageFile = readFileSync('./package.json');
+  const settings = JSON.parse(packageFile)['airtable-schema-generator'];
 
-  // Because nightmare stopped working :(
-  let schema = readSchemaFromFile();
-  if (!schema) {
-    const baseUrl = `https://airtable.com/login?continue=/${AIRTABLE_BASE_ID}/api/docs`;
-    const script = `copy(_.mapValues(application.tablesById, table => _.set(_.omit(table, ['sampleRows']),'columns',_.map(table.columns, item =>_.set(item, 'foreignTable', _.get(item, 'foreignTable.id'))))));`;
+  if (!settings || !settings.output || !settings.input || !settings.baseId) {
+    console.log(
+      "Couldn't find Input Folder Path, Output Folder Path and Base ID in Settings Object:"
+    );
+    console.log(settings);
     console.log(
-      `No Schema Found: Please create a schemaRaw.json file in the input folder.\n\n Navigate to this page in the browser: ${baseUrl} \n\nRun this script in the console:\n\n${script}\n\nThis will save the output to your clipboard. Open up schemaRaw.json and paste!`
+      "Please add appropriate values for 'baseId', 'input', and 'output' to package.json under 'airtable-schema-generator' key."
     );
-    return;
+    throw 'Invalid package.json settings';
   }
-
-  console.log('Retrieived Airtable Schema');
-
-  // Simplify the schema objects to just include table names and columns
-  let simplifiedSchema = simplifySchema(schema);
-
-  // Check for metadata input file
-  const schemaMetadata = getMetadata();
-
-  console.log(
-    'Found Metadata for tables: ' + Object.keys(schemaMetadata).toString()
-  );
-
-  // Generate outputs
-  generateSchemaFile(simplifiedSchema);
-  generateRequestFile(simplifiedSchema, schemaMetadata);
-  generateConstantsFile(simplifiedSchema);
-  generateAirtableFile(AIRTABLE_BASE_ID, DEFAULT_VIEW || 'Grid view');
-  console.log('Finished Generating Files!');
+  return {
+    outputFolder: settings.output,
+    inputFolder: settings.input,
+    baseId: settings.baseId,
```

Comment:
More of a question/nitpick: I wasn't aware `baseId` was being configured anywhere on the client side anymore? Is this supposed to be in `package.json`? 😅 

I wonder if it's easier to default it to `process.env.AIRTABLE_BASE_ID` (that's what our repo uses at least) even though it's not secret so there's no need for it to be hidden in an `.env` file. Otherwise it feels like there will be multiple sources of truth for `AIRTABLE_BASE_ID` for a single repo.. but anyway, nbd I think

Reply:
Ooh, this is a good point @ two sources of truth. I think in this case i'd prefer to keep base ID in 2 places rather than make it an environment variable, partially because I think I consider the schema generator a separate "app"